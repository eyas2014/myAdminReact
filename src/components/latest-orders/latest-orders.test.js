import React from 'react';
import LatestOrders from './latest-orders.jsx';
import enzyme from 'enzyme';

jest.mock('react-transition-group');
jest.mock('../../../lib/sparkline/jquery.sparkline.js');

var orders=[{
	orderId   : "OR9842",
	Item      : "call of duty",
	Status    : "shipped",
	Popularity: [1, 2, 10, -1, -2, 4, -6]
		},{
	orderId   : "OR9842",
	Item      : "Samsung Smart TV",
	Status    : "pending",
	Popularity: [1, 2, 5, -1, -2, 4, -6]
		},{
	orderId   : "OR9842",
	Item      : "iPhone 6 Plus",
	Status    : "delivered",
	Popularity: [1, 2, 5, -1, -2, 4, -6]
		},{
	orderId   : "OR9842",
	Item      : "Samsung Smart TV",
	Status    : "shipped",
	Popularity: [1, -1, -2, 4, -6, 2, 5]
		},{
	orderId   : "OR9842",
	Item      : "Samsung Smart TV",
	Status    : "processing",
	Popularity: [1, 2, -2, 4, -6, 5, -1]
		},{
	orderId   : "OR9842",
	Item      : "iPhone 6 Plus",
	Status    : "shipped",
	Popularity: [1, 2, 5, 4, -6, -1, -2]
		}];

document.body.innerHTML='<div  id="root"></div>';

$.isCalledWith=[];

const mountWrapper=enzyme.mount(<LatestOrders data={orders}></LatestOrders>, {attachTo: document.getElementById('root')});

it('should render correct htmls',()=>{
	expect(mountWrapper.html()).toMatchSnapshot();
});


it('chartjs should be called with correct arguments', ()=>{
	expect($.isCalledWith.length).toEqual(6);
});

it('should render correct number of orders', ()=>{
	expect(mountWrapper.find('.latest-orders main table tbody').children().length).toEqual(6);	
	expect(mountWrapper.find('.latest-orders main table tbody .inlineSpark').length).toEqual(6);
});