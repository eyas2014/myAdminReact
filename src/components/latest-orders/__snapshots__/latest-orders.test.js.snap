// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct htmls 1`] = `"<div class=\\"latest-orders box-shadow\\"><header class=\\"box-border-bottom\\"><div class=\\"float-right btn-group text-muted\\"><div class=\\"btn pull-right\\"><i class=\\"fas fa-minus\\"></i></div><div class=\\"btn pull-right\\"><i class=\\"fas fa-times\\"></i></div></div><div class=\\"btn\\"><span>Latest Orders</span></div></header><div class=\\"wrapper\\"><main class=\\"box-border-bottom\\"><div><table class=\\"w-100\\"><thead class=\\"box-border-bottom\\"><tr class=\\"font-weight-bold\\"><td>Order Id</td><td>Item</td><td>Status</td><td class=\\"text-center\\">Popularity</td></tr></thead><tbody><tr class=\\"box-border-top\\"><td class=\\"p-2 text-info\\"><span>OR9842</span></td><td class=\\"\\"><span>call of duty</span></td><td class=\\"\\"><span class=\\"bg-success badge p-2\\">shipped</span></td><td class=\\"text-center\\"><span class=\\"inlineSpark\\">1,2,10,-1,-2,4,-6</span></td></tr><tr class=\\"box-border-top\\"><td class=\\"p-2 text-info\\"><span>OR9842</span></td><td class=\\"\\"><span>Samsung Smart TV</span></td><td class=\\"\\"><span class=\\"bg-warning badge p-2\\">pending</span></td><td class=\\"text-center\\"><span class=\\"inlineSpark\\">1,2,5,-1,-2,4,-6</span></td></tr><tr class=\\"box-border-top\\"><td class=\\"p-2 text-info\\"><span>OR9842</span></td><td class=\\"\\"><span>iPhone 6 Plus</span></td><td class=\\"\\"><span class=\\"bg-danger badge p-2\\">delivered</span></td><td class=\\"text-center\\"><span class=\\"inlineSpark\\">1,2,5,-1,-2,4,-6</span></td></tr><tr class=\\"box-border-top\\"><td class=\\"p-2 text-info\\"><span>OR9842</span></td><td class=\\"\\"><span>Samsung Smart TV</span></td><td class=\\"\\"><span class=\\"bg-success badge p-2\\">shipped</span></td><td class=\\"text-center\\"><span class=\\"inlineSpark\\">1,-1,-2,4,-6,2,5</span></td></tr><tr class=\\"box-border-top\\"><td class=\\"p-2 text-info\\"><span>OR9842</span></td><td class=\\"\\"><span>Samsung Smart TV</span></td><td class=\\"\\"><span class=\\"bg-info badge p-2\\">processing</span></td><td class=\\"text-center\\"><span class=\\"inlineSpark\\">1,2,-2,4,-6,5,-1</span></td></tr><tr class=\\"box-border-top\\"><td class=\\"p-2 text-info\\"><span>OR9842</span></td><td class=\\"\\"><span>iPhone 6 Plus</span></td><td class=\\"\\"><span class=\\"bg-success badge p-2\\">shipped</span></td><td class=\\"text-center\\"><span class=\\"inlineSpark\\">1,2,5,4,-6,-1,-2</span></td></tr></tbody></table></div></main><footer><div><button class=\\"font-md px-2 py-1 bg-info text-white border-0\\">Place New Orders</button><button class=\\"font-md px-2 py-1 bg-grey float-right\\">Place New Orders</button></div></footer></div></div>"`;
